### DISCUSSION TITLE
wire:key is always needed?

### DISCUSSION THREAD

**Discussion by tmaeg** [livewire/livewire#8462] [1. Help]
[The document](https://livewire.laravel.com/docs/troubleshooting#adding-wirekey) says, `Any time you have a loop in your Blade templates using something like @foreach, you need to add wire:key to the opening tag of the first element within the loop`. So, it's inappropriate to omit wire:key even if the key is an array index, right?

```blade
<ul>
    @foreach ($errors->all() as $i => $error)
        <li wire:key="error-{{ $i }}">{{ $error }}</li>
    @endforeach
</ul>
```

**Comment by EriksonEnterprises** [comment]
I think it's good practice to do so, but usually I use: 

```wire:key="{{ $loop->index }}"```

Usually works with that

You can sometimes leave it out completely but then you risk Livewire losing it's tracking of where things are on the screen when you do dynamic updates

If you have nested foreach loops, or other loops on the same page, you can always do something like:

```wire:key="nestedName-{{ $loop->index }}"```

That way it won't collide with each other lol
